src/interfaces/sampler_interface.jl

```diff
@@ -0,0 +1,142 @@
+#################
+# The sampler interface
+#
+# In this file, we define the interface for general samplers and specific
+# samplers for scattering processes.
+#################
+
+"""
+Abstract base type for general sampler.
+
+!!! note "Sampler interface"
+
+    Functions to be implemented
+
+    ```Julia
+    CustomSampler <: AbstractSampler{Float64}
+
+    Base.eltype  # type of the generated samples
+    QEDevents._weight
+    Base.size  # Return the dimensionality of the sampler (number of outgoing four-momenta per sample).
+    is_exact
+    QEDevents._rand!(rng::AbstractRNG, ::AbstractSampler, x::AbstractVector{T}) where {T}
+
+    ```
+    
+    optional
+
+    ```Julia
+    _post_processing
+    ```
+
+"""
+abstract type AbstractSampler <: AbstractComputationSetup end
+
+@inline Base.eltype(::AbstractSampler) = Float64
```

Comment:
Is this implementation useful for anything? Wouldn't it lead to someone implementing a `CustomSampler`, forgetting to implement this, but then code still working but with a wrong element type?

Reply:
This is necessary to correctly allocate Vectors for results of a sampler, e.g. as seen here (cf line 118ff): 

```julia
function rand(rng::AbstractRNG, smplr::AbstractSampler)
    return _rand!(rng, smplr, Vector{eltype(smplr)}(undef, size(smplr, 1)))
end
```

Reply:
I wasn't questioning whether the function is needed at all, only whether this default implementation of it is needed for AbstractSamplers

Reply:
Ah, ok, understood. I think this is a design choice, whether an `AbstractSampler` returns a vector of floats or a vector of four-momenta, and we thought, the former is more convenient.